mbed_host_tests/host_tests_plugins/module_copy_mps2.py

```diff
@@ -17,96 +17,80 @@
-        new_file_lines = [] # New configuration file lines (entries)
-
-        # Check each line of the image configuration file
-        try:
-            with open(image_path, 'r') as file:
-                for line in file:
-                    if re.search('^TOTALIMAGES', line):
-                        # Check number of total images, should be 1
-                        new_file_lines.append(re.sub('^TOTALIMAGES:[\t ]*[\d]+', 'TOTALIMAGES: 1', line))
-                    elif re.search('; - %s[\n\r]*$'% MBED_SDK_TEST_STAMP, line):
-                        # Look for test suite entries and remove them
-                        pass    # Omit all test suite entries
-                    elif re.search('^IMAGE[\d]+FILE', line):
-                        # Check all image entries and mark the ';'
-                        new_file_lines.append(';' + line)   # Comment non test suite lines
-                    else:
-                        # Append line to new file
-                        new_file_lines.append(line)
-        except IOError as e:
-            return False
-
-        # Add new image entry with proper commented stamp
-        new_file_lines.append('IMAGE0FILE: %s    ; - %s\r\n'% (image0file_path, MBED_SDK_TEST_STAMP))
-
-        # Write all lines to file
-        try:
-            with open(image_path, 'w') as file:
-                for line in new_file_lines:
-                    file.write(line),
-        except IOError:
-            return False
-
-        return True
+    def mps2_bin_copy(self, image_path, destination_disk):
+        """! mps2 copy method for "mbed enabled" devices.
+            this copies the binary always as mbed.bin
+        @param image_path Path to binary file to be flashed
+        @param destination_disk Path to destination (mbed mount point)
 
-    def mps2_select_core(self, disk, mobo_config_name=""):
-        """ Function selects actual core
-        """
-        # TODO: implement core selection
-        pass
+        @details this uses default copy binary of the platform to copy the file.
 
-    def mps2_switch_usb_auto_mounting_after_restart(self, disk, usb_config_name=""):
-        """ Function alters configuration to allow USB MSD to be mounted after restarts
+        @return Returns True if copy (flashing) was successful
         """
-        # TODO: implement USB MSD restart detection
-        pass
-
-    # Plugin interface
-    name = 'HostTestPluginCopyMethod_MPS2'
-    type = 'CopyMethod'
-    capabilities = ['mps2']
-    required_parameters = ['image_path', 'destination_disk']
+        result = True
+        destination_path = os.path.join(destination_disk,"mbed.bin")
+        copy_command = 'cp' if os.name == 'posix' else 'copy'
+        result = self.run_command([copy_command, image_path, destination_path], shell=True)
+        if result and os.name == 'posix':
+            result = result and self.run_command('sync','-f',destination_path, shell=True)
```

Comment:
If copy fails we don't need a sync https://github.com/ARMmbed/htrun/pull/157/files always does a sync.